cheroot/connections.py

```diff
@@ -155,15 +158,29 @@ def get_conn(self, server_socket):
 
             # Wait for the next tick to occur.
             return None
-
-        try:
-            # See if we have a new connection coming in.
-            rlist.remove(ss_fileno)
-        except ValueError:
-            # No new connection, but reuse existing socket.
-            conn = socket_dict[rlist.pop()]
-        else:
-            conn = server_socket
+        finally:
+            for sock in socket_dict:
+                self._selector.unregister(sock)
```

Comment:
I'm not yet familiar with selectors, could you please tell me what happens with sockets between `unregister()` and the following `register()` at some point in time? Since the previous major contribution caused serious regressions, I'd like to be extra careful now...

Reply:
I'm not quite sure what you're asking here - are you saying we should be tracking events from sockets throughout their whole lifetime? i.e. as a connection is added to `self.connections`, we begin listening for events on it, and then as it's removed, we `unregister` it?

I haven't used sockets a lot before, so guidance/another solution from anyone else is very welcome

Reply:
I was just wondering how things work because I haven't used this API before. But that's fine if you don't have a detailed answer.